src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -204,11 +208,57 @@ SDCPUThread::onBeginFetch()
     if (!_cpuPtr->requestInstAddrTranslation(fetch_req, getThreadContext(),
         [this](Fault f, const RequestPtr& r) {onPCTranslated(f, r);})) {
         panic("The CPU rejected my fetch translate request and I haven't been "
-            "programmed to know how to continue!!!");
+              "programmed to know how to continue!!!");
         // TODO reschedule a fetch for the future?
     }
 }
 
+
+void
+SDCPUThread::onDataAddrTranslated(Fault fault, const RequestPtr& req,
+                                  InflightInst& inst, bool write)
+{
+    if (fault != NoFault) {
+        // TODO handle fault.
+        panic("SDCPU not able to handle data translation faults yet!");
+    }
+
+    // TODO set the effective address
```

Comment:
BTW, usually effective address is considered the virtual address of the access and is computed from the register+immediate offset. This is the physical address (usually).

You should be able to do memory disambiguation on the *effective address*. You shouldn't have to use the physical address unless switching address spaces is not serializing (it is).

Reply:
But you said synonyms were a problem? This was just me noting that I had to take care of it at some point (TODO).